### PR TITLE
 anonymousIdOptions added in JS SDK

### PR DESCRIPTION
**Pull request by ishachopra0401**
## Description of the change

Added anonymousIdOptions in JS SDK

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by MoumitaM — APPROVED** [review]

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/javascript-sdk-enhancements.mdx [hunk 1]
```diff
@@ -81,7 +81,7 @@ Currently, the RudderStack queue handles approximately 500 hits per 30ms and ens
 
 <div class="warningBlock">
   
-  <strong>This feature is now deprecated and will be sunset by the end of March 2022</strong>.
+  <strong>This feature is deprecated and RudderStack no longer provides its support</strong>.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/javascript-sdk-enhancements.mdx:84]:
```suggestion
  <strong>This feature is deprecated and no longer supported by RudderStack</strong>.
```

Comment (ameypv-rudder) [reply stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/javascript-sdk-enhancements.mdx:84]: Reads better, IMO.

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx [hunk 2]
```diff
@@ -579,6 +581,32 @@ rudderanalytics.load(
 For more information on the consent managers supported by RudderStack, refer to <a href="https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/consent-managers/">this section</a>.
 </div>
 
+#### Capture anonymous ID automatically
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:584]:
```suggestion
#### Capturing `anonymousId` automatically 
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx [hunk 3]
```diff
@@ -579,6 +581,32 @@ rudderanalytics.load(
 For more information on the consent managers supported by RudderStack, refer to <a href="https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/consent-managers/">this section</a>.
 </div>
 
+#### Capture anonymous ID automatically 
+
+The JavaScript SDK provides `anonymousIdOptions` object in the [`load()`](https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/#the-options-parameter) call options. It automatically captures the anonymous ID from a source and sets it as Rudderstack’s anonymous ID.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:586]:
```suggestion
The JavaScript SDK provides `anonymousIdOptions` object in the [`load()`](https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/#the-options-parameter) call options. It automatically captures the anonymous ID from a source and sets it as Rudderstack’s `anonymousId`.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx [hunk 4]
```diff
@@ -579,6 +581,32 @@ rudderanalytics.load(
...
+
+The structure of `anonymousIdOptions` object is shown below:
+
+```javascript
+{
+  autoCapture: {
+    enabled: true,
+    source: "segment",
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:594]: Do we need to mention "Segment" here? Can we use some other source? @MoumitaM

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx [hunk 5]
```diff
@@ -579,6 +581,32 @@ rudderanalytics.load(
...
+  }
+}
+```
+
+| Parameter  | Type  | Description    |   
+| :--------| :---- | :----------------| 
+| `enabled` | Boolean  | Identifies if the auto-capturing of anonymous ID is enabled. |
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:601]:
```suggestion
| `enabled` | Boolean  | Determines whether the auto-capturing of the anonymous ID is enabled. | 
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx (hunk 4 above)
Comment (MoumitaM) [reply stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:594]: As of now only segment is supported

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx [hunk 6]
```diff
@@ -579,6 +581,32 @@ rudderanalytics.load(
...
+```
+
+| Parameter  | Type  | Description    |   
+| :--------| :---- | :----------------| 
+| `enabled` | Boolean  | Identifies if the auto-capturing of anonymous ID is enabled. | 
+| `source`  | String   | Identifies the external source of the anonymous ID. | 
+
+<div class="infoBlock">
+  
+  If Rudderstack’s anonymous ID is already set in your browser, <code class="inline-code">anonymousIdOptions</code> will not take effect.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:606]:
```suggestion
  If the RudderStack `anonymousId` is already set in your browser, then <code class="inline-code">anonymousIdOptions</code> will not take effect.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx [hunk 7]
```diff
@@ -96,6 +97,7 @@ It includes the following details:
 | `useBeacon`           | Boolean                             | Defaults to `false`. If set to `true`, the SDK sends the event payloads via the  `navigator.sendBeacon()` utility. |
 | `beaconQueueOptions`  | [**BeaconQueueOpts**](https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/javascript-sdk-enhancements/#beaconqueueopts) | Refer to the [**BeaconQueueOpts**](https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/javascript-sdk-enhancements/#beaconqueueopts) section. |
 | `cookieConsentManager` | Object       |  Refer to the [`cookieConsentManager`](#cookieconsentmanager) section for more information. |
+| `anonymousIdOptions` | Object       |  Refer to the [Capture anonymous ID automatically](#capture-anonymous-id-automatically) section for more information. |
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/index.mdx:100]:
```suggestion
| `anonymousIdOptions` | Object       |  Refer to the [Capturing `anonymousId` automatically](#capture-anonymous-id-automatically) section for more information. |
```

**Comment by ameypv-rudder** [comment]
@MoumitaM to notify the doc team when this doc can be merged and released.

**Review by ameypv-rudder — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**